src/pages/workspace/companyCards/assignCard/ConfirmationStep.tsx

```diff
@@ -35,23 +35,32 @@ function ConfirmationStep({policyID, backTo}: ConfirmationStepProps) {
     const [assignCard] = useOnyx(ONYXKEYS.ASSIGN_CARD);
 
     const data = assignCard?.data;
-    const cardholderName = PersonalDetailsUtils.getPersonalDetailByEmail(data?.email ?? '')?.displayName ?? '';
-
+    const cardholderName = getPersonalDetailByEmail(data?.email ?? '')?.displayName ?? '';
+    const parts = backTo?.split('/');
+    const membersIndex = parts?.indexOf('members') ?? -1;
+    const workspaceMemberAccountID = parts?.[membersIndex + 1] ?? '';
```

Comment:
This is a hacky solution. We don't want to assume that the string `members` is always followed by. the account id. Can we instead extract the account id from `assignCard?.data` first value?

Reply:
I meant to use the first valid value from `assignCard?.data`. You can try use `useRef` or `usePrevious` hooks.

PS: Only store the first valid value i.e. check that `assignedData` is not `undefined` (e.g. onyx data is not loaded yet)

Reply:
I prefer to avoid that approach. Can we instead pass an extra param to the `WORKSPACE_EXPENSIFY_CARD_ISSUE_NEW` route and use it for comparison with the selected account id

Reply:
Let's not do that as that info is not part of the card data

Reply:
No need for a new function. You can add the param before calling  `getUrlWithBackToParam`. Please refer to `SETTINGS_2FA` or `SETTINGS_EXIT_SURVEY_RESPONSE`